### PR TITLE
[DRAFT] Improve documentation for bootstrapping and manager deployment

### PR DESCRIPTION
**Pull request by scoopex**
This change depends on https://github.com/osism/node-image/pull/84

### CODE REVIEW

File: docs/guides/deploy-guide/manager.md [hunk 1]
```diff
@@ -43,7 +43,16 @@ on the seed node. The deployment of the seed node itself is documented in the
 cd environments/manager
 ```
 
-### Step 1: Create operator user
+### Step 1: Configure deployment branch.
```
Comment (berendt) [inline docs/guides/deploy-guide/manager.md:46]: I would prefer to not have this as a separate step. It is sufficient to have main there by default and it will just work. This can be added somewhere else.

File: docs/guides/deploy-guide/manager.md [hunk 2]
```diff
@@ -112,16 +123,18 @@ in the Ansible documentation.
...
+
+
+To verify the proper creation of the operator user, use the private key file `id_rsa.operator`. Make
 sure you purge all keys from ssh-agent identity cache using `ssh-add -D`. You can print the list
 using `ssh-add -l`. The list should be empty.
 
 ```
 ssh-add -D
-ssh -o IdentitiesOnly=yes -i id_rsa.operator dragon@testbed-manager
+ssh -o IdentitiesOnly=yes -i  ../../secrets/id_rsa.operator dragon@YOUR_MANAGER_NODE
```
Comment (berendt) [inline docs/guides/deploy-guide/manager.md:134]: At the point the id_rsa.operator file is available in environments/manager and should be used. We cannot assume that the secrets directory is available on the seed node.

File: docs/guides/deploy-guide/manager.md [hunk 3]
```diff
@@ -167,41 +180,42 @@ This is recommended.
    ./run.sh reboot
    ```
 
-### Step 4: Deploy the manager service
+### Step 5: Deploy the manager service
 
-1. Transfer the configuration repository.
+1. Push your local configuration repo state of your seed node to to the git server
```
Comment (berendt) [inline docs/guides/deploy-guide/manager.md:185]: This is already part of "Creating a new configuration repository" and is not necessary here. There is a pointer to the "Creating a new configuration repository" chapter at the beginning of this chapter.

File: docs/guides/deploy-guide/manager.md [hunk 4]
```diff
@@ -167,41 +180,42 @@ This is recommended.
...
    ```
 
-### Step 5: Set vault password on the manager service
+### Step 6: Set vault password on the manager service
 
 Finally, the Ansible Vault password is made known on the manager node. Before that, log in to the manager node
 with the `dragon` user.
 
 ```
+ssh -o IdentitiesOnly=yes -i  ../../secrets/id_rsa.operator dragon@YOUR_MANAGER_NODE
```
Comment (berendt) [inline docs/guides/deploy-guide/manager.md:216]: Use id_rsa.operator instead of ../../secrets/id_rsa.operator. The private key is available at this point.

File: docs/guides/deploy-guide/provisioning.md [hunk 5]
```diff
@@ -17,9 +17,35 @@ control plane manually. An ISO image is provided for this purpose which automati
...
+* Configure a specific root password
+* Change other characteristics of the setup
+  * Templates
+  * Partitioning
+  * Packages
+  * ...
+* Develop new standard images
+
+The procedures fpr building custom images are described
+[osism/node-image](https://github.com/osism/node-image/blob/main/README.md) repository.
```
Comment (berendt) [inline docs/guides/deploy-guide/provisioning.md:47]: are described in the ..

File: docs/guides/operations-guide/index.md [hunk 6]
```diff
@@ -5,6 +5,47 @@ sidebar_position: 30
 
 # Operations Guide
 
+## General usage of osism commands
```
Comment (berendt) [inline docs/guides/operations-guide/index.md:8]: Move this into the manager/index.md file.

**Review by berendt — CHANGES_REQUESTED** [review]

File: docs/guides/upgrade-guide/manager.md [hunk 7]
```diff
@@ -27,8 +28,8 @@ adjustments are necessary
        use a version smaller v2. The v2 of Gilt is not yet usable.
 
        ```
-       gilt overlay  # you have to do this 2x
-       gilt overlay
+       # for releases < 7.0.0 : gilt overlay && gilt overlay
+       make sync
```
Comment (berendt) [inline docs/guides/upgrade-guide/manager.md:32]: Have a look at https://github.com/osism/osism.github.io/blob/main/docs/guides/deploy-guide/services/ceph.mdx?plain=1#L69-L86. This makes it easier for the reader.

File: docs/guides/deploy-guide/manager.md (hunk 1 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md:46]: I automatically used a different branch because my repo is configured to force a review of changes. Presumably this will also be the case in many productive environments, e.g. for compliance reasons from the outset. 
I have now marked the step as optional and added a clear note.

File: docs/guides/deploy-guide/manager.md (hunk 2 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md:134]: Added a placeholder instead

File: docs/guides/deploy-guide/manager.md (hunk 3 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md:185]: I agree, removed it

File: docs/guides/deploy-guide/manager.md (hunk 1 above)
Comment (berendt) [reply docs/guides/deploy-guide/manager.md:46]: It's still confusing for newcomers. I would prefer to have a section in the configuration guide about some more advanced use cases, like working with branched configuration repositories.

File: docs/guides/deploy-guide/manager.md (hunk 4 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md:216]: Same as above.

File: docs/guides/deploy-guide/provisioning.md (hunk 5 above)
Comment (scoopex) [reply docs/guides/deploy-guide/provisioning.md:47]: Fixed

File: docs/guides/operations-guide/index.md (hunk 6 above)
Comment (scoopex) [reply docs/guides/operations-guide/index.md:8]: Done, also beautified the image of on that page a bit.

File: docs/guides/upgrade-guide/manager.md (hunk 7 above)
Comment (scoopex) [reply docs/guides/upgrade-guide/manager.md:32]: Nice! Added it and made the legacy description a bit more verbose.

File: docs/guides/deploy-guide/manager.md (hunk 1 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md]: Done, moved to https://github.com/osism/osism.github.io/blob/manager_install_and_bootstrap/docs/guides/configuration-guide/manager.md#working-with-git-branches
Added a reference link to the Deployguide Page.

File: docs/guides/deploy-guide/bootstrap.md [hunk 8]
```diff
@@ -22,78 +22,131 @@ The guide for this can be found in the section [Provisioning of bare-metal nodes
...
+| Groupname             | Description                                           |
+|-----------------------|-------------------------------------------------------|
+| `[generic]`           | Roles which do not have any service specific settings |
+| `[manager]`           | Roles for the manager nodes                           |
+| `[monitoring]`        | The log, metric and alerting infrastructure roles     |
+| `[control]`           | Openstack contoller roles                             |
+| `[compute]`           | Openstack compute nodes roles                         |
+| `[ceph-control]`      | Ceph mon servers                                      |
+| `[ceph-resource]`     | Ceph OSDs, ...                                        |
+| `[ceph-rgw:children]` | Rados Gateway roles                                   |
```
Comment (scoopex) [inline docs/guides/deploy-guide/bootstrap.md:46]: Does it make sense to describe the purpose of the groups with a few words to
help newbies to make the right decisions?

File: docs/guides/deploy-guide/bootstrap.md [hunk 9]
```diff
@@ -114,41 +167,45 @@ the nodes are already ready for use.
      to host\r", "unreachable": true}
      ```
 
-7. Refresh facts.
+## Bootstrap the nodes
+
+1. Refresh facts.
```
Comment (scoopex) [inline docs/guides/deploy-guide/bootstrap.md:172]: A newcomer may ask: What exactly is this step responsible for and what purpose does it serve?

I think it would be good if we could link to a documentary here that goes into this.

File: docs/guides/operations-guide/manager/facts.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+---
+sidebar_label: Facts
+---
+
+# Facts
+
+TODOs:
```
Comment (scoopex) [inline docs/guides/operations-guide/manager/facts.md:7]: I think it would be a good idea to add a short explanation here.

File: docs/guides/operations-guide/manager/apply.md [hunk 11]
```diff
@@ -34,3 +38,56 @@ $ osism apply
...
+ok: [st01-mgmt-r01-u30]
+
+...
+...
+
+TASK [osism.commons.operator : Set password of operator user] ******************
+ok: [st01-mgmt-r01-u30]
+
+TASK [osism.commons.operator : Set root password like operator password] *******
+ok: [st01-mgmt-r01-u30]
```
Comment (berendt) [inline docs/guides/operations-guide/manager/apply.md:61]: This task is not part of the operator role and the PR will not be merged. We added the password feature in the osism.commons.user role.

File: docs/guides/operations-guide/manager/apply.md [hunk 12]
```diff
@@ -34,3 +38,56 @@ $ osism apply
 | iscsi                                                            | kolla          |
 ...
 ```
+
+## Apply a role
+
+```
+$ osism apply operator -l st01-mgmt-r01-u30
```
Comment (berendt) [inline docs/guides/operations-guide/manager/apply.md:45]: Do not use internal hostnames for samples.

File: docs/guides/operations-guide/manager/apply.md [hunk 13]
```diff
@@ -34,3 +38,56 @@ $ osism apply
...
+  gather_facts: false
+
+  tasks:
+    - name: Wipe partitions
+      become: true
+      ansible.builtin.shell: |
+        wipefs --all "{{ item }}"
+        dd if=/dev/zero of="{{ item }}" bs=1 count=4096
+      changed_when: false
+      with_items: "{{ devices }}"
```
Comment (berendt) [inline docs/guides/operations-guide/manager/apply.md:91]: While touching this sync this sample with the play https://github.com/osism/testbed/blob/main/environments/custom/playbook-wipe-partitions.yml.

File: docs/guides/deploy-guide/manager.md [hunk 14]
```diff
@@ -112,13 +116,15 @@ in the Ansible documentation.
...
+
+
+To verify the proper creation of the operator user, use the private key file `id_rsa.operator`. Make
 sure you purge all keys from ssh-agent identity cache using `ssh-add -D`. You can print the list
 using `ssh-add -l`. The list should be empty.
 
 ```
 ssh-add -D
-ssh -o IdentitiesOnly=yes -i id_rsa.operator dragon@testbed-manager
+ssh -o IdentitiesOnly=yes -i  PATH_TO_YOUR_SECRETS/id_rsa.operator dragon@YOUR_MANAGER_NODE
```
Comment (berendt) [inline docs/guides/deploy-guide/manager.md:127]: PATH_TO_YOUR_SECRETS is not required. The id_rsa.operator file is available at this location when running the manager seed steps.

File: docs/guides/deploy-guide/bootstrap.md [hunk 15]
```diff
@@ -102,53 +152,56 @@ the nodes are already ready for use.
...
    ```
    osism apply facts
    ```
 
-8. Bootstrap.
+2. Bootstrap nodes
 
    ```
-   osism apply bootstrap
+   osism apply bootstrap -l 'all:!manager'
```
Comment (berendt) [inline docs/guides/deploy-guide/bootstrap.md:177]: No. That's wrong. Run the bootstrap for all nodes, include the manager node.

File: docs/guides/deploy-guide/bootstrap.md [hunk 16]
```diff
@@ -102,53 +152,56 @@ the nodes are already ready for use.
...
+{"changed": false, "msg": "Connection timed out.", "unreachable": true}
      ```
 
    * System has already been rebooted and is not accessible via the network.
 
      ```
-     fatal: [net003]: UNREACHABLE! => {"changed": false, "msg": "EOF on stream;
-     last 100 lines received:\nssh: connect to host 10.15.0.33 port 22: No route
-     to host\r", "unreachable": true}
+fatal: [net003]: UNREACHABLE! =>
```
Comment (berendt) [inline docs/guides/deploy-guide/bootstrap.md:162]: Why is this necessary?

File: docs/guides/deploy-guide/bootstrap.md [hunk 17]
```diff
@@ -102,53 +152,56 @@ the nodes are already ready for use.
    * System is currently rebooting and is not yet accessible via network.
 
      ```
-     fatal: [net003]: UNREACHABLE! => {"changed": false, "msg": "Connection timed
-     out.", "unreachable": true}``
+fatal: [net003]: UNREACHABLE! =>
```
Comment (berendt) [inline docs/guides/deploy-guide/bootstrap.md:155]: Why is this necessary?

File: docs/guides/configuration-guide/manager.mdx [hunk 18]
```diff
@@ -63,11 +64,48 @@ In the example, OSISM release 7.0.5 is used.
...
+:::info
+
+For example, for compliance and security reasons, many organizations prefer to prepare changes to
+production systems on dedicated Git branches, roll them out to the production environment
+using the 4-eyes control principle and then finally transfer them to the `main` branch through a
+review and release process.
+
+A typical scenario is the [Manager Upgrade](../upgrade-guide/manager.mdx).
+
+OSISM offers the option of using specific GIT branches on the Manager.
```
Comment (berendt) [inline docs/guides/configuration-guide/manager.mdx:82]: Use Git.

File: docs/guides/configuration-guide/manager.mdx [hunk 19]
```diff
@@ -22,8 +22,8 @@ In the example, OSISM release 7.0.5 is used.
 1. Set the new manager version in the configuration repository.
 
    ```
-   MANAGER_VERSION=7.0.5
```
Comment (berendt) [inline docs/guides/configuration-guide/manager.mdx:25]: Keep MANAGER_VERSION

File: .github/workflows/markdown-link-check.json [hunk 20]
```diff
@@ -2,6 +2,10 @@
     "ignorePatterns": [
       {
         "pattern": "^(?!http(s)?:\/\/.*)|^(http(s)?:\/\/.*((osism.xyz)|(in-a-box.cloud)))(:?[0-9]+)?(\/.*)?$"
+      },
+      {
+        "pattern": "^https:\/\/www.wireguard.com$"
```
Comment (berendt) [inline .github/workflows/markdown-link-check.json:7]: This is not part of this PR. Do a proper rebase.

File: docs/guides/configuration-guide/manager.mdx [hunk 21]
```diff
@@ -41,10 +41,11 @@ In the example, OSISM release 7.0.5 is used.
...
    If Gilt is not installed via the `requirements.txt` of the manager environment it is
    important to use a version smaller v2. The v2 of Gilt is not yet usable.
-
    ```
-   gilt overlay  # you have to do this 2x, this is not a copy & paste error
-   gilt overlay
+   python3 -m venv venv
+   source venv/bin/activate
+   pip install -r requirements.txt
+     gilt overlay && gilt overlay
```
Comment (berendt) [inline docs/guides/configuration-guide/manager.mdx:48]: Use make sync.

File: docs/guides/configuration-guide/manager.mdx [hunk 22]
```diff
@@ -63,11 +64,48 @@ In the example, OSISM release 7.0.5 is used.
    workflows for changes to the configuration repository, only a generic example for Git.
 
    ```
-   git commit -a -s -m "manager: use OSISM version 7.0.5"
+   git commit -a -s -m "manager: use OSISM version ${UPGRADE_RELEASE?}"
```
Comment (berendt) [inline docs/guides/configuration-guide/manager.mdx:67]: Use MANAGER_VERSION

File: docs/guides/operations-guide/manager/reconceiler.md [hunk 23]
```diff
@@ -0,0 +1,21 @@
+---
+sidebar_label: Reconceiler
+---
+
+# Reconceiler
+
+TODOs:
```
[the remark this answers is not in the document]
Comment (berendt) [reply docs/guides/operations-guide/manager/reconceiler.md:7]: It's a reconciler and not a reconceiler.

**Review by berendt — CHANGES_REQUESTED** [review]

File: docs/guides/configuration-guide/manager.mdx (hunk 22 above)
Comment (scoopex) [reply docs/guides/configuration-guide/manager.mdx:67]: ok, renamed variable

File: docs/guides/operations-guide/manager/reconceiler.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (scoopex) [reply docs/guides/operations-guide/manager/reconceiler.md]: Renamed

File: .github/workflows/markdown-link-check.json (hunk 20 above)
Comment (scoopex) [reply .github/workflows/markdown-link-check.json]: Removed, just added for getting rid of the recurring problems with that domain.
After that i created the PR for the new link checker.

File: docs/guides/deploy-guide/manager.md (hunk 14 above)
Comment (scoopex) [reply docs/guides/deploy-guide/manager.md]: The user is probably located in the "<base-dir>/environments/manager" directory at this point. I wanted to make it clearer with this adjustment that the file is not located in the directory. On closer inspection, however, this is not true, because it is already the case on the seed node.
Removed "PATH_TO_YOUR_SECRETS/" now.

File: docs/guides/configuration-guide/manager.mdx (hunk 19 above)
Comment (scoopex) [reply docs/guides/configuration-guide/manager.mdx:25]: renamed.

File: docs/guides/operations-guide/manager/apply.md (hunk 11 above)
Comment (scoopex) [reply docs/guides/operations-guide/manager/apply.md]: Removed it

File: docs/guides/deploy-guide/manager.md (hunk 14 above)
Comment (berendt) [reply docs/guides/deploy-guide/manager.md]: The id_rsa.operator file is available in environments/manager when seeding the manager.

File: docs/guides/deploy-guide/bootstrap.md (hunk 15 above)
Comment (scoopex) [reply docs/guides/deploy-guide/bootstrap.md]: My reasoning was that at this point of the installation the "bootstrap" play was already executed as part of the manager deployment and is therefore rather unnecessary here. But basically it shouldn't do any harm. I have therefore now removed the restriction.

File: docs/guides/configuration-guide/manager.mdx (hunk 18 above)
Comment (scoopex) [reply docs/guides/configuration-guide/manager.mdx:82]: done

File: docs/guides/configuration-guide/manager.mdx (hunk 21 above)
Comment (scoopex) [reply docs/guides/configuration-guide/manager.mdx:48]: I added a extra tab item here because the "make sync" target ist not available for oder releases.

File: docs/guides/deploy-guide/bootstrap.md (hunk 17 above)
Comment (scoopex) [reply docs/guides/deploy-guide/bootstrap.md]: The build process handles this as some kind of javascript snippet: https://github.com/osism/osism.github.io/actions/runs/9548256571/job/26315015614

See "message": "Could not parse expression with acorn"

It seemed that it went away, bug now the problem is back :-(
Reverted now the example output to the original output.
I currently have no good idea what to do to make it work again :-)

**Comment by scoopex** [comment]
Replaced by smaller PRs, dropped
